### PR TITLE
[OpenCL] Registers AvgPoolGrad using SYCL kernel

### PR DESCRIPTION
**Pull request by jeffmak**

### CODE REVIEW

File: tensorflow/core/kernels/avgpooling_op.cc [hunk 1]
```diff
@@ -556,4 +556,207 @@ REGISTER_KERNEL_BUILDER(Name("AvgPoolGrad")
...
+  std::vector<int32> stride_;
+  Padding padding_;
+  TensorFormat data_format_;
+};
+
+REGISTER_KERNEL_BUILDER(Name("AvgPoolGrad")
+                            .Device(DEVICE_SYCL)
+                            .TypeConstraint<float>("T")
+                            .HostMemory("orig_input_shape")
+                            .Label("cudnn"),
```
Comment (lukeiwanski) [inline tensorflow/core/kernels/avgpooling_op.cc:758]: is that label needed here?

File: tensorflow/core/kernels/pooling_ops_common.h [hunk 2]
```diff
@@ -68,6 +68,69 @@ struct PoolParameters {
   TensorFormat data_format;
 };
 
+#ifdef TENSORFLOW_USE_SYCL
```
Comment (lukeiwanski) [inline tensorflow/core/kernels/pooling_ops_common.h:71]: This change set might cause conflict as it appears in two pull requests. ( https://github.com/lukeiwanski/tensorflow/pull/99 )

**Review by lukeiwanski — CHANGES_REQUESTED** [review]
There is quite big overlap between this PR and https://github.com/lukeiwanski/tensorflow/pull/99 .. it might be better to create one PR for both.

**Comment by jeffmak** [comment]
Closing PR, refer to #105.

**Pull request closed without merging**